lib/puppet/provider/service/redhat.rb

```diff
@@ -10,7 +10,7 @@
 
   commands :chkconfig => "/sbin/chkconfig", :service => "/sbin/service"
 
-  defaultfor 'os.family' => :redhat
+  defaultfor 'os.name' => :amazon, 'os.release.major' => ["2017"]
```

Comment:
I found that [Amazon Linux 1 may report the major version as 2018](https://github.com/puppetlabs/puppetlabs-puppet_agent/pull/692#discussion_r1439934276), so maybe add `2018` to the array and the `notdefaultfor` in the `systemd` provider.

I think this overall change may be breaking on RedHat (or derivative) for:
1. systems (excluding SUSE 10/11, AL1 and RHEL4-6) which don't use systemd. Though that [list is basically empty](https://distrowatch.com/search.php?ostype=All&category=All&origin=All&basedon=Red+Hat&notbasedon=None&desktop=All&architecture=All&package=All&rolling=All&isosize=All&netinstall=All&language=All&defaultinit=Not+systemd&status=Active#simple).
2. systems where `systemd` is not available due to the `confine :true` line above, see 93ca5f0ea2b99e8c7a1b5619fa02a6c56f202b33. Going back in history it looks like it was previously updated to support puppet running in a `chroot` environment  a7b9d0e78787bb39d545c48afe32119e20d70dc8, and before that, the line was added for `Debian` in c36f1e9e6c6 (so maybe the entire backwards compat issue doesn't matter?)

Reply:
Thanks for the review Josh. I updated the Amazon Linux statements (their versioning scheme can be confusing . . . ).

Given what you've found, I agree that platforms potentially affected by this change seem very limited, probably esoteric, and definitely not officially supported anyway.